src/pystack/_pystack/mem.cpp

```diff
@@ -368,7 +368,7 @@ CorefileRemoteMemoryManager::copyMemoryFromProcess(remote_addr_t addr, size_t si
     StatusCode ret = getMemoryLocationFromCore(addr, &offset_in_file);
 
     if (ret == StatusCode::SUCCESS) {
-        if (static_cast<size_t>(offset_in_file) > d_corefile_size) {
+        if (static_cast<size_t>(offset_in_file + size) > d_corefile_size) {
```

Comment:
This should be written a bit differently to avoid integer overflow bugs, I think. It's a bit pedantic, but I think we should do:
```suggestion
        if (size > d_corefile_size || static_cast<size_t>(offset_in_file) > d_corefile_size - size) {
```

I'm sure this doesn't matter at all for 64 bit builds, but it might for 32 bit builds.